collection-loader.js

```diff
@@ -1,42 +1,104 @@
+    .then((response) => response.results);
 }
 
 // Ugly. This function updates all the items in place.
 // However, this is way more readable than a pure version
-function updateItemsInPlace(client, depth, items, {storyFields, storyLimits}) {
-  const collections = items.filter(item => item && item.type == "collection");
-
-  if(depth == 0 || collections.length == 0)
-    return Promise.resolve();
-
-  return loadCollectionItems(client, collections, {storyFields, storyLimits})
-    .then(collectionSlugToCollection => {
-      collections.forEach(collection => {
-        collection.summary = get(collectionSlugToCollection, [collection.slug, "summary"], '')
-        collection.items = get(collectionSlugToCollection, [collection.slug, "items"])
-      });
-      return updateItemsInPlace(client, depth - 1, flatMap(collections, collection => collection.items), {storyFields, storyLimits})
-    })
+function updateItemsInPlace(
+  client,
+  depth,
+  items,
+  {storyFields, storyLimits, defaultNestedLimit, nestedCollectionLimit}
+) {
+  const collections = items.filter((item) => item && item.type == 'collection');
+
+  if (depth == 0 || collections.length == 0) return Promise.resolve();
+
+  return loadCollectionItems(client, collections, {
+    storyFields,
+    storyLimits,
+    defaultNestedLimit,
+  }).then((collectionSlugToCollection) => {
+    collections.forEach((collection) => {
+      collection.summary = get(
+        collectionSlugToCollection,
+        [collection.slug, 'summary'],
+        ''
+      );
+      collection.items = get(collectionSlugToCollection, [
+        collection.slug,
+        'items',
+      ]);
+    });
+
+    collections.forEach((collection) =>
+      collection.items
+        .filter((item) => item && item.type === 'collection')
+        .forEach((nestedChildCollection, index) => {
+          console.log(nestedCollectionLimit);
+          if (
+            nestedCollectionLimit[
+              get(collection, ['associated-metadata', 'layout'])
+            ]
+          ) {
+            nestedChildCollection.childCollectionLimit =
+              nestedCollectionLimit[
+                get(collection, ['associated-metadata', 'layout'])
+              ][index];
+          }
+        })
+    );
+
+    return updateItemsInPlace(
+      client,
+      depth - 1,
+      flatMap(collections, (collection) => collection.items),
+      {storyFields, storyLimits, defaultNestedLimit}
+    );
+  });
 }
 
-function loadNestedCollectionData(client, collection, {depth, storyFields, storyLimits}) {
-  return updateItemsInPlace(client, depth, collection.items, {storyFields, storyLimits})
-    .then(() => collection);
+function loadNestedCollectionData(
+  client,
+  collection,
+  {depth, storyFields, storyLimits, defaultNestedLimit, nestedCollectionLimit}
+) {
+  return updateItemsInPlace(client, depth, collection.items, {
+    storyFields,
+    storyLimits,
+    defaultNestedLimit,
+    nestedCollectionLimit,
```

Comment:
I don't see this in the related malibu pr ?

Reply:
https://github.com/quintype/malibu/pull/736